test/e2e/velero_utils.go

```diff
@@ -192,24 +202,104 @@ func CheckRestorePhase(ctx context.Context, veleroCLI string, restoreName string
-func VeleroRestore(ctx context.Context, veleroCLI string, restoreName string, backupName string) error {
-	restoreCmd := exec.CommandContext(ctx, veleroCLI, "create", "restore", restoreName, "--from-backup", backupName, "--wait")
+func VeleroRestore(ctx context.Context, veleroCLI string, veleroNamespace string, restoreName string, backupName string) error {
+	restoreCmd := exec.CommandContext(ctx, veleroCLI, "--namespace", veleroNamespace, "create", "restore", restoreName,
+		"--from-backup", backupName, "--wait")
+
+	restoreCmd.Stdout = os.Stdout
+	restoreCmd.Stderr = os.Stderr
 	fmt.Printf("restore cmd =%v\n", restoreCmd)
 	err := restoreCmd.Run()
 	if err != nil {
 		return err
 	}
-	return CheckRestorePhase(ctx, veleroCLI, restoreName, velerov1api.RestorePhaseCompleted)
+	return CheckRestorePhase(ctx, veleroCLI, veleroNamespace, restoreName, velerov1api.RestorePhaseCompleted)
+}
+
+func VeleroInstall(ctx context.Context, veleroNamespace string, cloudProvider string, objectStoreProvider string, useVolumeSnapshots bool,
+	cloudCredentialsFile string, bslBucket string, bslPrefix string, bslConfig string, vslConfig string) error {
+
+	if cloudProvider != "kind" {
+		if objectStoreProvider != "" {
+			return errors.New("For cloud platforms, object store plugin cannot be overridden") // Can't set an object store provider that is different than your cloud
+		}
+		objectStoreProvider = cloudProvider
+	} else {
+		if objectStoreProvider == "" {
+			return errors.New("No object store provider specified - must be specified when using kind as the cloud provider") // Gotta have an object store provider
+		}
+	}
+	err := EnsureClusterExists(ctx)
+	if err != nil {
+		return errors.WithMessage(err, "Failed to ensure kubernetes cluster exists")
+	}
+	veleroInstallOptions, err := GetProviderVeleroInstallOptions(objectStoreProvider, cloudCredentialsFile, bslBucket, bslPrefix, bslConfig, vslConfig, getProviderPlugins(objectStoreProvider))
+	if err != nil {
+		return errors.WithMessagef(err, "Failed to get Velero InstallOptions for plugin provider %s", objectStoreProvider)
+	}
+	veleroInstallOptions.UseRestic = !useVolumeSnapshots
+	veleroInstallOptions.Namespace = veleroNamespace
+	err = InstallVeleroServer(veleroInstallOptions)
+	if err != nil {
+		return errors.WithMessagef(err, "Failed to install Velero in cluster")
+	}
+	return nil
+}
+
+func VeleroUninstall(ctx context.Context, client *kubernetes.Clientset, extensionsClient *apiextensionsclient.Clientset,
+	veleroNamespace string) error {
+	// TODO - replace with invocation of "velero uninstall" when that becomes available
+	err := client.CoreV1().Namespaces().Delete(ctx, veleroNamespace, metav1.DeleteOptions{})
+	if err != nil {
+		return errors.WithMessagef(err, "Uninstall failed removing Velero namespace")
+	}
+	crds, err := extensionsClient.ApiextensionsV1().CustomResourceDefinitions().List(ctx, metav1.ListOptions{
+		LabelSelector: "component=velero",
```

Comment:
Thanks to #3447, we can re-use the definition of the label selector from the install package, so we only have 1 canonical place to change it. In particular, [this line](https://github.com/vmware-tanzu/velero/commit/6455350940eff352762c9b89e756fa23fe8add3b#diff-c6ec5e408c7f64e1ab2f86dcd1887b2486ee370c2142403a0f7c0a8f67349f5eR33) shows how to go from a map to a string.

Reply:
Done